app/models/terms_of_service.rb

```diff
@@ -6,29 +6,50 @@
 #
 #  id                   :bigint(8)        not null, primary key
 #  changelog            :text             default(""), not null
+#  effective_date       :date
 #  notification_sent_at :datetime
 #  published_at         :datetime
 #  text                 :text             default(""), not null
 #  created_at           :datetime         not null
 #  updated_at           :datetime         not null
 #
 class TermsOfService < ApplicationRecord
-  scope :published, -> { where.not(published_at: nil).order(published_at: :desc) }
-  scope :live, -> { published.limit(1) }
+  scope :published, -> { where.not(published_at: nil).order(effective_date: :desc) }
+  scope :live, -> { published.where('effective_date < now()').limit(1) }
```

Comment:
This fails considering existing ToS records with no `effective_date` and causes multiple issues, like old records with no `effective_date` being listed first. Maybe a `COALESCE(effective_date, published_at)` would make the most sense here…